modules/aws-emr-cluster/main.tf

```diff
@@ -25,31 +25,42 @@ resource "aws_emr_cluster" "emr-cluster" {
     key_name                          = var.key_pair_name
   }
 
-
-  master_instance_group {
-    name          = var.master_instance_group_name
-    instance_type = var.master_instance_type
-    # NOTE: value must be 1 or 3
-    instance_count = var.master_group_instance_count
-    # Spot Instance definition
-    bid_price = var.master_bid_price
-    ebs_config {
-      size                 = var.master_ebs_size
-      type                 = var.master_ebs_type
-      volumes_per_instance = var.master_ebs_volumes_count
+  master_instance_fleet {
+    instance_type_configs {
+      instance_type = var.master_instance_type
+      ebs_config {
+        size                 = var.master_ebs_size
+        type                 = var.master_ebs_type
+        volumes_per_instance = var.master_ebs_volumes_count
+      }
     }
+    target_on_demand_capacity = 1
+    name          = var.master_instance_group_name
   }
-  core_instance_group {
-    name           = var.core_instance_group_name
-    instance_type  = var.core_instance_type
-    instance_count = var.core_group_instance_count
-    # Spot Instance definition
-    bid_price = var.core_bid_price
-    ebs_config {
-      size                 = var.core_ebs_size
-      type                 = var.core_ebs_type
-      volumes_per_instance = var.core_ebs_volumes_count
+
+  core_instance_fleet {
+    instance_type_configs {
+      bid_price_as_percentage_of_on_demand_price = 100
+      ebs_config {
+        size                 = var.core_ebs_size
+        type                 = var.core_ebs_type
+        volumes_per_instance = var.core_ebs_volumes_count
+      }
+      instance_type     = var.core_instance_type
+      weighted_capacity = 1
+    }
+    launch_specifications {
+      spot_specification {
+        allocation_strategy      = "capacity-optimized"
+        block_duration_minutes   = 0
```

Comment:
Do you think it's worth having variables for `block_duration_minutes` and `timeout_duration_minutes`?

Reply:
`block_duration_minutes` is interesting because essentially it's a more deterministic version of spot instances, where you're guaranteed the instance will be around for x minutes, but you'll also be guaranteed to lose the instance at the end of that duration.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/emr_cluster#block_duration_minutes

I'm not sure that's optimal for hbase to lose all the spot instances around the same time (most likely), so we might not want that yet.

Totally agree that `timeout_duration_minutes` should be a variable.